### PR TITLE
add time windows support in artifact trash

### PR DESCRIPTION
**Pull request by wy65701436**
support with time window to filter and delete item in artifact trash

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/pkg/artifactrash/dao/dao.go [hunk 1]
```diff
@@ -16,9 +16,9 @@ type DAO interface {
 	// Delete the artifact trash specified by ID
 	Delete(ctx context.Context, id int64) (err error)
 	// Filter lists the artifact that needs to be cleaned
-	Filter(ctx context.Context) (arts []model.ArtifactTrash, err error)
+	Filter(ctx context.Context, timeWindow time.Time) (arts []model.ArtifactTrash, err error)
```
Comment (reasonerjt) [inline src/pkg/artifactrash/dao/dao.go:19]: I think the var name is misleading, timewindow means a range of time.

File: src/pkg/artifactrash/manager.go [hunk 2]
```diff
@@ -32,9 +33,9 @@ type Manager interface {
 	// Delete ...
 	Delete(ctx context.Context, id int64) (err error)
 	// Filter ...
-	Filter(ctx context.Context) (arts []model.ArtifactTrash, err error)
+	Filter(ctx context.Context, timeWindow int64) (arts []model.ArtifactTrash, err error)
```
Comment (reasonerjt) [inline src/pkg/artifactrash/manager.go:36]: We should clarify the window here is hour

File: src/pkg/artifactrash/dao/dao.go (hunk 1 above)
Comment (wy65701436) [reply src/pkg/artifactrash/dao/dao.go]: updated.

File: src/pkg/artifactrash/manager.go (hunk 2 above)
Comment (wy65701436) [reply src/pkg/artifactrash/manager.go:37]: updated.

**Pull request merged**